dfc.providerportal.findanapprenticeship/src/dfc.providerportal.findanapprenticeship/Storage/AzureBlobStorageClient.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+
+namespace Dfc.Providerportal.FindAnApprenticeship.Storage
+{
+    public class AzureBlobStorageClient : IBlobStorageClient
+    {
+        private readonly BlobContainerClient _blobContainerClient;
+
+        public AzureBlobStorageClient(AzureBlobStorageClientOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            _blobContainerClient = new BlobContainerClient(options.ConnectionString, options.BlobContainerName);
+            _blobContainerClient.CreateIfNotExists(PublicAccessType.None);
```

Comment:
I'm not crazy about doing IO in constructors but if this is the simplest way then I guess it's ok

Reply:
I prefer this as it limits the number of times this is executed and given it that the class' purpose is a client for the container, it seems to me like a valid dependency to enforce during construction.